Failure to parse empty html tags (also when nested)

We are failing to parse empty matched tags:


```

➜  src javadoc ComplexJavadocExample.java
Loading source file ComplexJavadocExample.java...
Constructing Javadoc information...
Building index for all the packages and classes...
Standard Doclet version 21.0.6+7-Ubuntu-122.04.1
Building tree for all the packages and classes...
Generating ./ComplexJavadocExample.html...
ComplexJavadocExample.java:4: warning: use of default constructor, which does not provide a comment
public class ComplexJavadocExample {
       ^
Generating ./package-summary.html...
Generating ./package-tree.html...
Generating ./overview-tree.html...
Building index for all classes...
Generating ./allclasses-index.html...
Generating ./allpackages-index.html...
Generating ./index-all.html...
Generating ./search.html...
Generating ./index.html...
Generating ./help-doc.html...
1 warning

➜  src cat ComplexJavadocExample.java
/**
 * <p>This is a test... <div><span><a></a></span></div>
 */
public class ComplexJavadocExample {
}
```

Found at https://github.com/checkstyle-GSoC25/javadoc-parser/issues/21

--- a/src/main/resources/JavadocParser.g4
+++ b/src/main/resources/JavadocParser.g4
@@ -97,7 +97,7 @@ voidElement
     : {isVoidTag()}? htmlTagStart
     ;
 
-tight: {!ParserUtility.isNonTightTag(_input, unclosedTagNameTokens)}? htmlTagStart htmlContent htmlTagEnd;
+tight: {!ParserUtility.isNonTightTag(_input, unclosedTagNameTokens)}? htmlTagStart htmlContent? htmlTagEnd;
 
 nonTight: htmlTagStart nonTightHtmlContent?;
 

--- a/src/test/java/AstTest.java
+++ b/src/test/java/AstTest.java
@@ -123,4 +123,9 @@ public class AstTest extends AbstractTestSupport {
     public void testNonTightTags6() throws IOException {
         verifyAst(getPath("htmlNonTight6.txt"), getPath("htmlNonTight6.javadoc"));
     }
+
+    @Test
+    public void testEmptyHtmlContent() throws IOException {
+        verifyAst(getPath("emptyHtmlContent.txt"), getPath("emptyHtmlContent.javadoc"));
+    }
 }

--- a/src/test/resources/emptyHtmlContent.javadoc
+++ b/src/test/resources/emptyHtmlContent.javadoc
@@ -1,0 +1,2 @@
+* <p>This is a test... <div><span><a></a></span></div>
+* <p><div><span><a></a></span></div>

--- a/src/test/resources/emptyHtmlContent.txt
+++ b/src/test/resources/emptyHtmlContent.txt
@@ -1,0 +1,75 @@
+'- javadoc
+   |- mainDescription
+   |  |- htmlElement
+   |  |  |- htmlTagStart
+   |  |  |  |- TOKEN[type: TAG_OPEN, text: <]
+   |  |  |  |- TOKEN[type: TAG_NAME, text: p]
+   |  |  |  '- TOKEN[type: TAG_CLOSE, text: >]
+   |  |  '- nonTightHtmlContent
+   |  |     '- TOKEN[type: TEXT, text: This is a test... ]
+   |  |- htmlElement
+   |  |  |- htmlTagStart
+   |  |  |  |- TOKEN[type: TAG_OPEN, text: <]
+   |  |  |  |- TOKEN[type: TAG_NAME, text: div]
+   |  |  |  '- TOKEN[type: TAG_CLOSE, text: >]
+   |  |  |- htmlContent
+   |  |  |  |- htmlTagStart
+   |  |  |  |  |- TOKEN[type: TAG_OPEN, text: <]
+   |  |  |  |  |- TOKEN[type: TAG_NAME, text: span]
+   |  |  |  |  '- TOKEN[type: TAG_CLOSE, text: >]
+   |  |  |  |- htmlContent
+   |  |  |  |  |- htmlTagStart
+   |  |  |  |  |  |- TOKEN[type: TAG_OPEN, text: <]
+   |  |  |  |  |  |- TOKEN[type: TAG_NAME, text: a]
+   |  |  |  |  |  '- TOKEN[type: TAG_CLOSE, text: >]
+   |  |  |  |  '- htmlTagEnd
+   |  |  |  |     |- TOKEN[type: TAG_OPEN, text: <]
+   |  |  |  |     |- TOKEN[type: TAG_SLASH, text: /]
+   |  |  |  |     |- TOKEN[type: TAG_NAME, text: a]
+   |  |  |  |     '- TOKEN[type: TAG_CLOSE, text: >]
+   |  |  |  '- htmlTagEnd
+   |  |  |     |- TOKEN[type: TAG_OPEN, text: <]
+   |  |  |     |- TOKEN[type: TAG_SLASH, text: /]
+   |  |  |     |- TOKEN[type: TAG_NAME, text: span]
+   |  |  |     '- TOKEN[type: TAG_CLOSE, text: >]
+   |  |  '- htmlTagEnd
+   |  |     |- TOKEN[type: TAG_OPEN, text: <]
+   |  |     |- TOKEN[type: TAG_SLASH, text: /]
+   |  |     |- TOKEN[type: TAG_NAME, text: div]
+   |  |     '- TOKEN[type: TAG_CLOSE, text: >]
+   |  |- TOKEN[type: NEWLINE, text: \n]
+   |  |- htmlElement
+   |  |  |- TOKEN[type: TAG_OPEN, text: <]
+   |  |  |- TOKEN[type: TAG_NAME, text: p]
+   |  |  '- TOKEN[type: TAG_CLOSE, text: >]
+   |  '- htmlElement
+   |     |- htmlTagStart
+   |     |  |- TOKEN[type: TAG_OPEN, text: <]
+   |     |  |- TOKEN[type: TAG_NAME, text: div]
+   |     |  '- TOKEN[type: TAG_CLOSE, text: >]
+   |     |- htmlContent
+   |     |  |- htmlTagStart
+   |     |  |  |- TOKEN[type: TAG_OPEN, text: <]
+   |     |  |  |- TOKEN[type: TAG_NAME, text: span]
+   |     |  |  '- TOKEN[type: TAG_CLOSE, text: >]
+   |     |  |- htmlContent
+   |     |  |  |- htmlTagStart
+   |     |  |  |  |- TOKEN[type: TAG_OPEN, text: <]
+   |     |  |  |  |- TOKEN[type: TAG_NAME, text: a]
+   |     |  |  |  '- TOKEN[type: TAG_CLOSE, text: >]
+   |     |  |  '- htmlTagEnd
+   |     |  |     |- TOKEN[type: TAG_OPEN, text: <]
+   |     |  |     |- TOKEN[type: TAG_SLASH, text: /]
+   |     |  |     |- TOKEN[type: TAG_NAME, text: a]
+   |     |  |     '- TOKEN[type: TAG_CLOSE, text: >]
+   |     |  '- htmlTagEnd
+   |     |     |- TOKEN[type: TAG_OPEN, text: <]
+   |     |     |- TOKEN[type: TAG_SLASH, text: /]
+   |     |     |- TOKEN[type: TAG_NAME, text: span]
+   |     |     '- TOKEN[type: TAG_CLOSE, text: >]
+   |     '- htmlTagEnd
+   |        |- TOKEN[type: TAG_OPEN, text: <]
+   |        |- TOKEN[type: TAG_SLASH, text: /]
+   |        |- TOKEN[type: TAG_NAME, text: div]
+   |        '- TOKEN[type: TAG_CLOSE, text: >]
+   '- TOKEN[type: EOF, text: <EOF>]

